example/lib/star_wars/planet_screen.dart

```diff
@@ -0,0 +1,207 @@
+  Widget build(BuildContext context) {
+    final state = Provider.of<AppState>(context);
+    final index = useMemoized(() {
+      if (state.isFetchingPlanets) {
+        return 0;
+      }
+
+      if (state.planetPage.results.isEmpty) {
+        return 1;
+      }
+
+      if (state.errorFetchingPlanets != null) {
+        return 2;
+      }
+
+      return 3;
+    }, [
+      state.isFetchingPlanets,
+      state.planetPage.results.isEmpty,
+      state.errorFetchingPlanets
+    ]);
+
+    return IndexedStack(
+      children: <Widget>[
+        Container(
+          alignment: Alignment.center,
+          child: const CircularProgressIndicator(),
+        ),
+        Container(
+          alignment: Alignment.center,
+          child: const Text('No planets found'),
+        ),
+        _Error(
+          errorMsg: state.errorFetchingPlanets,
+        ),
+        _PlanetList(),
+      ],
+      index: index,
+    );
+  }
+}
+
+class _Error extends StatelessWidget {
+  final String errorMsg;
+
+  const _Error({Key key, this.errorMsg}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final handler = Provider.of<_PlanetHandler>(context);
+    return Container(
+      alignment: Alignment.center,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          if (errorMsg != null) Text(errorMsg),
+          RaisedButton(
+            color: Colors.redAccent,
+            child: const Text('Try again'),
+            onPressed: () async {
+              await handler.fetchAndDispatch();
+            },
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _LoadPageButton extends HookWidget {
+  _LoadPageButton({this.next = true}) : assert(next != null);
+
+  final bool next;
+
+  @override
+  Widget build(BuildContext context) {
+    final handler = Provider.of<_PlanetHandler>(context);
+    final state = Provider.of<AppState>(context);
+    return RaisedButton(
+      child: next ? const Text('Next Page') : const Text('Prev Page'),
+      onPressed: () async {
+        final url = next ? state.planetPage.next : state.planetPage.previous;
+        await handler.fetchAndDispatch(url);
+      },
+    );
+  }
+}
+
+class _PlanetList extends HookWidget {
+  @override
+  Widget build(BuildContext context) {
+    final state = Provider.of<AppState>(context);
+    return ListView.builder(
+      itemCount: 1 + state.planetPage.results.length,
+      itemBuilder: (context, index) {
+        if (index == 0) {
+          return _PlanetListHeader();
+        }
+
+        final planet = state.planetPage.results[index - 1];
+        return ListTile(
+          title: Text(planet.name),
+        );
+      },
+    );
+  }
+}
+
+class _PlanetListHeader extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final state = Provider.of<AppState>(context);
+    return HookBuilder(builder: (context) {
+      final buttonAlignment = useMemoized(
```

Comment:
The `useMemoized` is overkill here too

It can be removed (and HookBuilder at the same time)